src/attributes.jl

```diff
@@ -1856,6 +1856,108 @@ function attribute_value_type(::ConstraintConflictStatus)
     return ConflictParticipationStatusCode
 end
 
+"""
+    UserDefinedFunction(name::Symbol, arity::Int) <: AbstractModelAttribute
+
+Set this attribute to register a user-defined function by the name of `name`
```

Comment:
Should this attribute be set or submitted?

Reply:
So I remembered why I made this a model attribute: we don't cache `AbstractSubmittable` in the `.model_cache`. That means, if you change the solver, we'd loose the user-defined function information and need to re-register it.

Currently, the only submittables are `LazyConstraint`, `HeuristicSolution` and `UserCut` which are set during a callback.